### PR TITLE
fix(cloudflare): split `Set-Cookie` header into multiple headers

### LINKED ISSUE #988: Set Cookie merged into one header (Cloudflare Workers)
### Environment

Cloudflare Workers entry

### Reproduction

Make proxy which sets more than 2 cookies 

### Describe the bug

When you set cookies all you cookies merged into one header, while it’s not allowed by RFC. You need to pass one header per cookie 

### Additional context

I gonna open PR and fix it, I will link it 

### Logs

_No response_

### PR DESCRIPTION
**Pull request by oleghalin**
### 🔗 Linked issue

#988

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Splits Set-Cookie for each upstream header provided. 

Resolves #988 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by oleghalin** [comment]
I suppose that these changes should be applied also to other presets where `.join(',')` used.

**Comment by oleghalin** [comment]
I checked codebase a little bit deeper and i think that it should be changed on unenv level and response.headers (which is HeadersObject) should implement Headers instead of `{ [key: string]: string | string[] | undefined };` and with this we will need to make such transformations on preset level.

File: src/runtime/entries/cloudflare.ts [hunk 1]
```diff
@@ -71,9 +73,20 @@ const baseURLModifier = (request: Request) => {
...
-  return Object.entries(headers).map(([k, v]) => [
-    k,
-    Array.isArray(v) ? v.join(",") : v,
-  ]);
+): Headers {
+  const result = new Headers();
+
+  for (const [k, v] of Object.entries(headers)) {
+    if (k === "set-cookie") {
+      for (const cookie of splitCookiesString(v)) {
```
Comment (pi0) [inline src/runtime/entries/cloudflare.ts:81]: We have recently introduced [same util](https://github.com/unjs/h3/blob/80f2c611918a17d1c303636924ce39ea7f246489/src/utils/internal/cookies.ts#L10) in h3. It might be one less dependency and bundle size if we expose it from h3 package.

Comment (oleghalin) [reply src/runtime/entries/cloudflare.ts:81]: Thanks, what is proper way to sync versions of h3 and current nitro? Should we wait for h3 version tag or?

Comment (pi0) [reply src/runtime/entries/cloudflare.ts:81]: Sure, releasing next version until this weekend. (at the same time thinking about your other comments, we need to really fix header normalization across entries)

Comment (oleghalin) [reply src/runtime/entries/cloudflare.ts:81]: As i see normalize function looks same for all other entries, but i think something is different on provider level, i suppose that some of same do cookie splitting themselves. We can drop normalize function to utils and use it every entry until make changes with type change for `HadersObject`.

Comment (pi0) [reply src/runtime/entries/cloudflare.ts:81]: Indeed there are slight behavior diferences. Also we never properly tested header behavior on [nitro-deploys](https://github.com/unjs/nitro-deploys) 😓

Comment (oleghalin) [reply src/runtime/entries/cloudflare.ts:81]: Hello, @pi0. Do you have any plans for releasing new versions soon? Gonna continue work on module and project, but this issue blocks me testing any env except local one :)

**Comment by oleghalin** [comment]
Hello, @pi0. I also pushed to this PR change for not caching 404 responses if its asset directory, can you check it too please

**Comment by oleghalin** [comment]
Hello, @pi0 any updates on it? Issue with Set-Cookie header and 404 caching its making Nuxt absolutely unusuable with Cloudlare Workers

**Comment by Stupremee** [comment]
Hey @oleghalin,
I just saw this PR because I have the same problems with Nuxt on CF Workers, which is soo annoying.

I noticed that the new `h3` version used in nitro now exports the `splitCookeString` function, so maybe you can remove the `set-cookie-parser` dependency and this PR can move forward.

**Comment by oleghalin** [comment]
Hello, @pi0. I just just updated my PR and dropped third-party dep

**Comment by oleghalin** [comment]
Hey, @pi0. I suppose that it makes many problems for all users who use CF workers to deploy, can you take a look on it? It solves 2 big problems with caching and cookies

**Comment by jhd4tmr** [comment]
+1 I had the same bug that bothered me for days，it's not just about ```nitro.routeRules```, ```proxyRequest``` has the same bug

**Comment by Hebilicious** [comment]
@oleghalin This is on my radar, apologies for the delay. Can you provide me with a reproduction ? I would like to see if this behaviour affects the other cloudflare presets too.

**Review by Hebilicious — CHANGES_REQUESTED** [review]
This doesn't include any tests or reproduction, so this is hard to test.

**Comment by oleghalin** [comment]
Not sure why you need reproduction, Set-Cookie header should be merged into one because browsers read each Set-Cookie header and set 1 cookie per each Set-Cookie header, as soon as there is only 1 header delimited by coma it will write only first cookie. As I know Vercel and node server parse it on their end and thats why it works, but it doesn't work for CF.

**Comment by Hebilicious** [comment]
Thanks for the additional information and the PR.
A reproduction is useful to test the proposed changes. Without one, it takes more maintainer time to review and get things merged.

File: src/runtime/entries/cloudflare.ts [hunk 2]
```diff
@@ -3,27 +3,38 @@ import {
...
-    });
-  } catch {
-    // Ignore
+  const url = new URL(event.request.url);
+  const id = decodeURIComponent(
+    withLeadingSlash(withoutTrailingSlash(url.pathname))
+  );
+
+  // Fetch public assets from KV only
+  if (isPublicAssetURL(id)) {
```
Comment (Hebilicious) [inline src/runtime/entries/cloudflare.ts:27]: I agree this needs to happen, but it should be handled in a different PR (https://github.com/unjs/nitro/pull/1236/files)

**Review by Hebilicious — COMMENTED** [review]
This is absolutely a necessary change! 

- The check for publicAsset needs to happen in another PR https://github.com/unjs/nitro/pull/1236 (feedback welcome)

- This affects multiple preset and needs to be applied everywhere
